src/main/java/org/folio/inventory/dataimport/handlers/matching/AbstractMatchEventHandler.java

```diff
@@ -63,12 +70,58 @@ public CompletableFuture<DataImportEventPayload> handle(DataImportEventPayload d
 
   private CompletableFuture<Boolean> doMatching(DataImportEventPayload dataImportEventPayload, MappingMetadataDto mappingMetadataDto,
-                                                MatchingParametersRelations matchingParametersRelations) {
+                                                MatchingParametersRelations matchingParametersRelations, Context context) {
     dataImportEventPayload.getContext().put(MAPPING_PARAMS, mappingMetadataDto.getMappingParams());
     dataImportEventPayload.getContext().put(MATCHING_RELATIONS,
       Json.encode(matchingParametersRelations.getMatchingRelations()));
 
-    return MatchingManager.match(dataImportEventPayload);
+    return MatchingManager.match(dataImportEventPayload)
+      .thenCompose(matchedLocal -> {
+        if (isConsortiumAvailable()) {
+          return matchCentralTenantIfNeeded(dataImportEventPayload, matchedLocal, context, mappingMetadataDto, matchingParametersRelations);
+        }
+        return CompletableFuture.completedFuture(matchedLocal);
+      });
+  }
+
+  private CompletableFuture<Boolean> matchCentralTenantIfNeeded(DataImportEventPayload dataImportEventPayload, boolean isMatchedLocal, Context context,
+                                                                MappingMetadataDto mappingMetadataDto, MatchingParametersRelations matchingParametersRelations) {
+    return consortiumService.getConsortiumConfiguration(context)
+      .toCompletionStage().toCompletableFuture()
+      .thenCompose(consortiumConfiguration -> {
+        if (consortiumConfiguration.isPresent() && !consortiumConfiguration.get().getCentralTenantId().equals(context.getTenantId())) {
+          LOGGER.debug("matchCentralTenantIfNeeded:: Start matching on central tenant with id: {}", consortiumConfiguration.get().getCentralTenantId());
+          String localMatchedInstance = dataImportEventPayload.getContext().get(getEntityType().value());
+          preparePayloadBeforeConsortiumProcessing(dataImportEventPayload, consortiumConfiguration.get(), mappingMetadataDto, matchingParametersRelations);
+          return MatchingManager.match(dataImportEventPayload)
+            .thenCompose(isMatchedConsortium -> {
+              dataImportEventPayload.setTenant(context.getTenantId());
+              if (isMatchedConsortium && isMatchedLocal) {
+                LOGGER.warn("matchCentralTenantIfNeeded:: Found multiple results during matching on local tenant: {} and central tenant: {} ",
+                  consortiumConfiguration.get().getCentralTenantId(), context.getTenantId());
+                return CompletableFuture.failedFuture(new MatchingException(String.format(FOUND_MULTIPLE_ENTITIES, context.getTenantId(), consortiumConfiguration.get().getCentralTenantId())));
+              }
+              if (StringUtils.isEmpty(dataImportEventPayload.getContext().get(getEntityType().value()))) {
+                LOGGER.debug("matchCentralTenantIfNeeded:: Matched on local tenant: {}", context.getTenantId());
```

Comment:
should this log line be shown when we match on local tenant?
Thinking we would see log lines that shows there is a match on central+local but no log for match on local only.